app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -49,6 +49,8 @@ def set_user_by_token(mapping=nil)
       if devise_warden_user && devise_warden_user.tokens[@client_id].nil?
         @used_auth_by_token = false
         @resource = devise_warden_user
+        # REVIEW: Why are we bothering to create an auth token here? It won't
+        # get used anywhere by the looks of it...?
         @resource.create_new_auth_token
```

Comment:
I guess it was previously logged with devise and doesn't have a token, maybe @lynndylanhurley can give some hint

Reply:
I tried just commenting the line out and re-running tests, and it led to a failing test that was added in [PR #434](https://github.com/lynndylanhurley/devise_token_auth/pull/434/files#diff-41be446a925979b6d8df8a6361c2ef48). I repeated this on the `master` branch and it led to the same failing test.

Upon closer inspection of [PR #434](https://github.com/lynndylanhurley/devise_token_auth/pull/434/files), I'm not sure that the tests defined actually test the feature added (limiting concurrent devices/tokens). Furthermore, it looks like testing maximum concurrent tokens happens inside the block that tests the standard devise auth functionality rather than token auth.

Can anyone confirm or refute this?

Reply:
Thinking about this further, I'm having a hard time understanding a use case for this. 

This block of code is only relevant if `DeviseTokenAuth.enable_standard_devise_support = true`, and if an existing user has been authenticated with warden/devise.  Under these circumstances, it seems like not only is there no need to create a token, but the controller response  _**should not**_ contain any token auth related headers either.

Does anyone know if a use case for returning token auth headers despite authenticating via warden/standard devise?

Reply:
@Evan-M please feel free to remove the tests for now or do what you need to do to move the PR forward!!

Reply:
Don't remove tests that should be working despite it's using single or multi auth providers. I don't know why this line, but if you can't find the reason, just remove the comment above

Reply:
Sorry @MaicolBen ....just trying to help @Evan-M move this PR forward as it's been open for a while and would be awesome to get in the project!

Reply:
@zachfeldman @MaicolBen: I am very reluctant to remove any tests! 

However, I am trying to better understand the intent of these tests, and the implementation code that satisfies them.  If warranted, I will re-write or re-factor some of these tests.

Admittedly, some of my questions/comments here are perhaps beyond the scope of this PR (allowing multiple auth methods per resource). 

Currently, I've added 11 commits to my [fork](https://github.com/hirepool/devise_token_auth/tree/multi_auth_methods_per_resource_rebased) of this PR (`13 files changed, 102 insertions(+), 40 deletions(-)`, according to `git diff --shortstat`). Additionally, there are prior the commits by @zachfeldman, @ram535ii, and others.  Personally, I would prefer to provide shorter, and easier to understand (i.e. review) PRs.

To this end, I plan on spending time this weekend to extract some of the commits from my [fork](https://github.com/hirepool/devise_token_auth/tree/multi_auth_methods_per_resource_rebased) into separate PRs.  I think most of these separate PRs will focus on cleaning up confusing or redundant code, and ensuring that certain tests do in fact fully test what they intend to.

Perhaps most crucially, I want to ensure everything remains backwards compatible!

Reply:
@Evan-M I totally agree that this PR is huge and it's pretty hard to move forward with such a monolith....looking forward to seeing your work!